src/components/main/files/upload/mod.rs

```diff
@@ -111,11 +146,66 @@ pub fn Upload<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
     })
 }
 
-async fn set_thumbnail_if_file_is_image(
-    file_storage: Storage,
-    filename_to_save: String,
-) -> Result<String, Box<dyn std::error::Error>> {
-    let item = file_storage.root_directory().get_item(&filename_to_save)?;
+
+
+async fn upload_file(file_storage: Storage, file_path: PathBuf) {
+    let filename = std::path::Path::new(&file_path)
+    .file_name()
+    .unwrap_or_else(|| std::ffi::OsStr::new(""))
+    .to_str()
+    .unwrap()
+    .to_string();
+
+    let local_path = Path::new(&file_path).to_string_lossy().to_string();
+    let mut filename_to_save = filename.clone();
+    let mut count_index_for_duplicate_filename = 1;
+    let mut file_storage = file_storage.clone();
+
+    loop {
+        match file_storage.put(&filename_to_save, &local_path).await {
```

Comment:
I think it might be best to check the current directly for the filename prior to uploading so when you do get the final name or if it doesnt exist you can then proceed to upload it than to loop over `Constellation::put`. Here is an example of what im doing https://github.com/Satellite-im/Warp/blob/102752d6e243c91364ee11b6417bdeaf9106122a/extensions/warp-rg-ipfs/src/store/direct.rs#L1331-L1357